### PR TITLE
Overlay Lighting Fixes Part 2

### LINKED ISSUE #5962: Non-functional Crusher lights
## Round ID: 
34711

## Testmerges: 
#5897, #5611

## Reproduction:
1. Take a proto-kinetic crusher out
2. Go somewhere dark
3. Try to turn on the light

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request

- [x] Fix Kinetic Crusher Light ( #5962  )
- [x] Ghosts can't see in the dark ( #5942  ) [ Explained ]
- [x] Hardsuit helmets glowing ( #5921 ) [ not going to fix that or anything relating hardsuits until #6034 is ported]
- [x] Ability to see in the dark when you have high contrast [ Explained ]
- [x] Fix swarmer light not turning back on
- [x] Fix pAI light not working correctly
- [x] desword glowing 

Fixes #5962 

## Why It's Good For The Game
Bugfixes good

## Changelog
:cl:
fix: Fixed kinetic Crusher Light not working
fix: Fixed desword glowing even if not wielded
fix: Fixed swarmer lights
fix: Fixed pAI lights
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
Its not the high contrast that let him see in the dark. Its apparently the night vision trait iirc

**Comment by Dejaku51** [comment]
> Its not the high contrast that let him see in the dark. Its apparently the night vision trait iirc

nope, i checked it and you can see in the dark on high contrast

**Comment by Dejaku51** [comment]
> Its not the high contrast that let him see in the dark. Its apparently the night vision trait iirc

so it is not a bug?

**Comment by KubeRoot** [comment]
Make sure to use the exact `Fixes #issue`/`Closes #issue` phrasing to have issues automatically close when the PR is merged (for the things you've actually fixed, at least, in case this one is merged before the checklist is finished too)

**Comment by ivanmixo** [comment]
> > Its not the high contrast that let him see in the dark. Its apparently the night vision trait iirc
> 
> so it is not a bug?

Makes night vision very OP

**Comment by BrumalLake** [comment]
i don't think you could even see anything more with nv before

**Comment by ivanmixo** [comment]
Syndie hardsuit light still uses old lighting

**Comment by SergeiKoralev** [comment]
Please verify that the ERT hardsuits work as intended with lights. The engineering one had no light at all last time I checked.

- Sergei

**Comment by Dejaku51** [comment]
> Please verify that the ERT hardsuits work as intended with lights. The engineering one had no light at all last time I checked.
> 
> * Sergei

as I said "not going to fix that or anything relating hardsuits until #6034 is ported"

File: code/modules/antagonists/swarmer/swarmer.dm [hunk 1]
```diff
@@ -679,10 +679,16 @@
...
 /mob/living/simple_animal/hostile/swarmer/proc/ToggleLight()
-	if(!light_range)
-		set_light_on(TRUE)
-	else
+	if(swarmer_flags & SWARMER_LIGHT_ON)
+		swarmer_flags = ~SWARMER_LIGHT_ON
 		set_light_on(FALSE)
+		if(!mind)
+			return
+		return
```
Comment (Archanial) [inline code/modules/antagonists/swarmer/swarmer.dm:687]: ?

File: code/modules/antagonists/swarmer/swarmer.dm [hunk 2]
```diff
@@ -679,10 +679,16 @@
...
+	swarmer_flags |= SWARMER_LIGHT_ON
+	set_light_on(TRUE)
+	if(!mind)
+		return
```
Comment (Archanial) [inline code/modules/antagonists/swarmer/swarmer.dm:691]: ?

File: code/modules/antagonists/swarmer/swarmer.dm [hunk 3]
```diff
@@ -679,10 +679,16 @@
...
-	if(!light_range)
-		set_light_on(TRUE)
-	else
+	if(swarmer_flags & SWARMER_LIGHT_ON)
+		swarmer_flags = ~SWARMER_LIGHT_ON
 		set_light_on(FALSE)
+		if(!mind)
+			return
+		return
+	swarmer_flags |= SWARMER_LIGHT_ON
```
Comment (Archanial) [inline code/modules/antagonists/swarmer/swarmer.dm:688]: You can toggle bitflags using ^= operator

**Comment by Archanial** [comment]
thank you ike very cool

**Comment by DatBoiTim** [comment]
I'd fix the hardsuit skinsuit issue. MODSuits won't be done for a bit. Even without additional fixes to port.

**Pull request merged**